src/components/home.js

```diff
@@ -0,0 +1,111 @@
+import { useEffect } from 'react';
+import { useNavigate, Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import getSymbolFromCurrency from 'currency-symbol-map';
+import { BsArrowRightCircle, BsCurrencyExchange } from 'react-icons/bs';
+import { fetchAllCurrencies, displaySingleCurrency, displayCurrencies } from '../redux/Api/fetchCRapi';
+import Nav from './navbar';
+import '../App.css';
+
+const Homepage = () => {
+  const currencies = useSelector((state) => state.currencies);
+  const dispatch = useDispatch();
+  const currenciesArray = Object.keys(currencies[0]).map((key) => ({
+    code: key,
+    name: currencies[0][key],
+  }));
+
+  const currencyArray = Object.keys(currencies[1]).map((key) => ({
+    code: key,
+    name: currencies[1][key],
+  }));
+
+  const dataToDisplayFunc = () => ((currencyArray.length !== 0) ? currencyArray : currenciesArray);
+  const dataToDisplay = dataToDisplayFunc();
+
+  const showPreferedCurrency = (e) => {
+    const code = e.target.value;
+    if (code === 'allc') {
+      dispatch(displayCurrencies());
+    } else {
+      dispatch(displaySingleCurrency(code));
+    }
+    // remove focus from the select element
+    e.target.blur();
+  };
+
+  useEffect(() => {
+    if (currenciesArray.length === 0) {
+      dispatch(fetchAllCurrencies());
+    }
+  }, []);
+
+  const navigate = useNavigate();
+
+  const redirectToCurrency = (code) => {
+    navigate(`/currency/${code}`);
+  };
+
+  return (
+    <>
+      <header>
+        <Nav text="Currencies Exchange">
+          <select onChange={showPreferedCurrency}>
+            {/* default options to be all currencies */}
+            <option value="allc">All Currencies</option>
+            {currenciesArray.map((currency) => (
+              <option key={currency.code} value={currency.code}>
+                {currency.name}
+              </option>
+            ))}
+          </select>
```

Comment:
- [Optional] In my opinion, we could have just displayed all currencies on the homepage without the select options. Next, we could have used ES6 JS filter() method to filter out currencies by name using a search bar in a stateless Filter component. This way, it'll be more easier for the user to search for a particular item(currency) by just introducing the name in the search bar.



Below is a screenshot of how the dropdown options occur on mobile screens. Not just responsive. Kindly make sure to fix the issue.



![Screenshot from 2022-10-13 03-41-48](https://user-images.githubusercontent.com/63087491/195486823-8aa111e1-dbf2-4170-99f9-be1a975c5b3e.png)